acs-apiclient/Droid/Activity/ExternalFlowWebActivity.cs

```diff
@@ -0,0 +1,130 @@
+            SetupWebView();
+        }
+
+        private void InitUri()
+        {
+            this.urlString = this.Intent.GetStringExtra(UrlParamKey);
+            if (String.IsNullOrEmpty(this.urlString))
+            {
+                throw new ArgumentException($"'url' must be provided in the Intent to start {typeof(ExternalFlowWebActivity)}");
+            }
+            this.uri = Android.Net.Uri.Parse(this.urlString);
+        }
+
+        private void SetupTitle(string newTitle)
+        {
+            TextView title = FindViewById<TextView>(Resource.Id.textview_titlebarWithExitAndRefreshButtons_title);
+            title.Text = newTitle;
+        }
+
+        private void SetupSyncButton()
+        {
+            EasyTintImageButton syncButton = FindViewById<EasyTintImageButton>(Resource.Id.easyTintImageButton_titlebarWithExitAndRefreshButtons_sync);
+            syncButton.SelectedTintColor = SelectedButtonTintColor;
+            syncButton.UnSelectedTintColor = UnselectedButtonTintColor;
+            syncButton.Click += (object sender, EventArgs e) => 
+            {
+                this.contentWebView.LoadUrl(this.urlString);
+            };
+        }
+
+        private void SetupCloseButton()
+        {
+            EasyTintImageButton closeButton = FindViewById<EasyTintImageButton>(Resource.Id.easyTintImageButton_titlebarWithExitAndRefreshButtons_close);
+            closeButton.SelectedTintColor = SelectedButtonTintColor;
+            closeButton.UnSelectedTintColor = UnselectedButtonTintColor;
+            closeButton.Click += (object sender, EventArgs e) => 
+            {
+                LoginController.Instance.UserCancelledLogin();
+                Finish();
+            };
+        }
+
+        private void SetupWebView()
+        {
+            this.contentWebView = FindViewById<WebView>(Resource.Id.webview_externalFlowWeb_content);
+            this.contentWebView.Settings.JavaScriptEnabled = true;
+            this.contentWebView.Settings.DomStorageEnabled = true;
+            this.contentWebView.SetWebViewClient(new ExternalFlowWebViewClient(this));
+            this.contentWebView.LoadUrl(this.urlString);
+        }
+        
+        public class ExternalFlowWebViewClient : WebViewClient
+        {
+            Activity activity;
+
+            public ExternalFlowWebViewClient(Activity activity)
+            {
+                this.activity = activity;
+            }
+            
+            /**
+            * Use shouldOverrideUrlLoading(WebView, WebResourceRequest) instead when the Android app's min SDK is increased to >= 24
+            * https://developer.android.com/reference/android/webkit/WebViewClient.html#shouldOverrideUrlLoading(android.webkit.WebView, android.webkit.WebResourceRequest)
+            */
+            public override bool ShouldOverrideUrlLoading(WebView webview, String url)
+            {
+                Console.WriteLine($"ShouldOverrideUrlLoading: url={url}");
```

Comment:
⚠️ this testing console log should be removed

Reply:
Agreed

Reply:
Done.

Reply:
Done.

Reply:
✅ Not done, but fixed in review commit